h2o-core/src/main/java/hex/Model.java

```diff
@@ -2432,8 +2432,21 @@ public boolean testJavaScoring(Frame data, Frame model_predictions, EasyPredictM
             boolean deleted = new File(filename).delete();
             if (!deleted) Log.warn("Failed to delete the file");
           }
-        }
 
+          if (! Arrays.equals(model_predictions.names(), genmodel.getOutputNames())) {
+            if (_parms._distribution == DistributionFamily.quasibinomial) {
+              Log.warn("Quasibinomial doesn't correctly return output names in MOJO"); 
+            } else if (genmodel.getModelCategory() == ModelCategory.Clustering && Arrays.equals(genmodel.getOutputNames(), new String[]{"cluster"})) {
```

Comment:
@michalk:  I am not sure this is correct.  @maurever went through all the trouble to make sure I use the correct names and here you said that is not correct?

Reply:
@wendycwong Definitely not consistent, PredictCsv returns "cluster" and H2O uses "predict".